src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/resource/PrisonApiResource.kt

```diff
@@ -5,22 +5,29 @@ import io.swagger.v3.oas.annotations.media.Content
 import io.swagger.v3.oas.annotations.media.Schema
 import io.swagger.v3.oas.annotations.responses.ApiResponse
 import io.swagger.v3.oas.annotations.tags.Tag
+import org.springframework.http.HttpStatus
 import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
 import org.springframework.web.bind.annotation.RestController
 import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.config.ErrorResponse
 import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.dto.prisonapi.NonAssociationDetails
 import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.service.NonAssociationsService
 
 @RestController
+@Validated
 @RequestMapping("/legacy/api", produces = [MediaType.APPLICATION_JSON_VALUE])
 @Tag(name = "Non-Associations", description = "Retrieve non association details from prison-api")
 class PrisonApiResource(
   val nonAssociationsService: NonAssociationsService,
 ) {
   @GetMapping("/offenders/{prisonerNumber}/non-association-details")
+  @PreAuthorize("hasRole('ROLE_NON_ASSOCIATIONS')")
```

Comment:
is this a role that's already set it up in hmpps-auth somewhere? the `hmpps-non-associations` client doesn't have it in dev (that's the only place i can check)

Reply:
I'll set it up

Reply:
Prison API endpoints often work off the relationship between a user and the prisoner rather than specific roles.  For that endpoint see https://github.com/ministryofjustice/prison-api/blob/fb3e3e69317a88512224778b28ff248f46bf1e35/src/main/java/uk/gov/justice/hmpps/prison/service/OffenderNonAssociationsService.java#L47.  The Prison API swagger docs go into more detail.